Modules/_datetimemodule.c

```diff
@@ -1980,8 +1989,30 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
                     goto Done;
                 }
             }
-
-            ntoappend = PyOS_snprintf(buf, sizeof(buf), "%04ld", year_long);
+#ifdef Py_STRFTIME_C99_SUPPORT
+            if (ch == 'F') {
+                item = PyTuple_GET_ITEM(timetuple, 1);
+                long month = PyLong_AsLong(item);
+                if (month == -1 && PyErr_Occurred()) {
+                    goto Done;
+                }
+                item = PyTuple_GET_ITEM(timetuple, 2);
+                long day = PyLong_AsLong(item);
+                if (day == -1 && PyErr_Occurred()) {
+                    goto Done;
+                }
+                ntoappend = PyOS_snprintf(buf, sizeof(buf), "%04ld-%02ld-%02ld",
+                                          year_long, month, day);
```

Comment:
There is potential buffer overflow if `month` or `day` exceed the 0-99 range.

One way to solve this -- increase the buffer size (it should be tripled). Other way -- write literal '-%m-%d' in the buffer.

BTW, we should check that `timetuple` is a tuple and has right size before the first use of `PyTuple_GET_ITEM()`.

Reply:
Ah the latter solution is so much better. Agreed on the error checks. Thanks again!